Extension/src/LanguageServer/settings.ts

```diff
@@ -350,51 +361,58 @@ export class CppSettings extends Settings {
+    public get workspaceSymbols(): string { return this.getAsString("workspaceSymbols"); }
+    public get exclusionPolicy(): string { return this.getAsString("exclusionPolicy"); }
+    public get refactoringIncludeHeader(): string { return this.getAsString("refactoring.includeHeader"); }
+    public get simplifyStructuredComments(): boolean { return this.getAsBoolean("simplifyStructuredComments"); }
+    public get doxygenGeneratedCommentStyle(): string { return this.getAsString("doxygen.generatedStyle"); }
+    public get doxygenGenerateOnType(): boolean { return this.getAsBoolean("doxygen.generateOnType"); }
+    public get commentContinuationPatterns(): (string | CommentPattern)[] {
+        const value: any = super.Section.get<any>("commentContinuationPatterns");
+        if (this.isArrayOfCommentContinuationPatterns(value)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp.commentContinuationPatterns");
+        return setting.default;
+    }
+    public get isConfigurationWarningsEnabled(): boolean { return this.getAsString("configurationWarnings").toLowerCase() === "enabled"; }
+    public get preferredPathSeparator(): string { return this.getAsString("preferredPathSeparator"); }
+    public get updateChannel(): string { return this.getAsString("updateChannel"); }
+    public get vcpkgEnabled(): boolean { return this.getAsBoolean("vcpkg.enabled"); }
+    public get addNodeAddonIncludePaths(): boolean { return this.getAsBoolean("addNodeAddonIncludePaths"); }
+    public get renameRequiresIdentifier(): boolean { return this.getAsBoolean("renameRequiresIdentifier"); }
+    public get filesExclude(): Excludes { return this.getAsExcludes("files.exclude"); }
+    public get defaultIncludePath(): string[] | undefined { return this.getArrayOfStringsWithUndefinedDefault("default.includePath"); }
+    public get defaultDefines(): string[] | undefined { return this.getArrayOfStringsWithUndefinedDefault("default.defines"); }
+    public get defaultDotconfig(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("default.dotConfig")); }
+    public get defaultMacFrameworkPath(): string[] | undefined { return this.getArrayOfStringsWithUndefinedDefault("default.macFrameworkPath"); }
+    public get defaultWindowsSdkVersion(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("default.windowsSdkVersion")); }
+    public get defaultCompileCommands(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("default.compileCommands")); }
+    public get defaultForcedInclude(): string[] | undefined { return this.getArrayOfStringsWithUndefinedDefault("default.forcedInclude"); }
+    public get defaultIntelliSenseMode(): string | undefined { return this.getAsStringOrUndefined("default.intelliSenseMode"); }
+    public get defaultCompilerPath(): string | null { return this.getAsString("default.compilerPath", true); }
```

Comment:
Why was `undefined` removed as a value?

Reply:
Because it should not be possible to get undefined for this setting.
```
                    "C_Cpp.default.compilerPath": {
                        "type": [
                            "string",
                            "null"
                        ],
                        "default": null,
                        "markdownDescription": "%c_cpp.configuration.default.compilerPath.markdownDescription%",
                        "scope": "machine-overridable"
                    },
```